src/engines/mod.rs

```diff
@@ -578,7 +580,11 @@ fn input_seek<'a, I: 'a + IoProvider>(es: *mut ExecutionState<'a, I>, handle: *m
         libc::SEEK_SET => SeekFrom::Start(offset as u64),
         libc::SEEK_CUR => SeekFrom::Current(offset as i64),
         libc::SEEK_END => SeekFrom::End(offset as i64),
-        _ => panic!("Unexpected \"whence\" parameter to fseek() wrapper: {}", whence),
+        _ => {
+                tt_warning!(es.status, "Unexpected \"whence\" parameter to fseek() wrapper: {}",
+                            whence);
+                return 0;
```

Comment:
I think it would make more sense to change `input_seek` to return a signed value. `fseek` returns `-1` and set `errno` to `EINVAL` when there is an invalid `whence` provided. I think it would make sense to return `-EINVAL` (e.g., `ssize_t` is signed). A negative value so that the caller knows there is an error and its value set to an error to know the reason if needed.